### PR TITLE
bpo-33262: Deprecate passing None for `s` to shlex.split()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue33262

### CODE REVIEW

**Comment by matrixise** [comment]
Hi @ZackerySpytz 

Thank you for your PR and your time.

**Review by matrixise — APPROVED** [review]

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Please add also an entry in the What's New document.

**Comment by ZackerySpytz** [comment]
I have made the requested changes; please review again.

**Review by pganssle — APPROVED** [review]

File: Doc/library/shlex.rst [hunk 1]
```diff
@@ -36,6 +36,8 @@ The :mod:`shlex` module defines the following functions:
       instance, passing ``None`` for *s* will read the string to split from
       standard input.
 
+   .. deprecated:: 3.8
+      Passing ``None`` for *s*.
```
Comment (pganssle) [inline Doc/library/shlex.rst]:
```suggestion
      Passing ``None`` for *s* will raise an exception in future Python versions.
```

This is a bit of nitpicking, which is why I reviewed as "Approved" here, but I think the current "deprecated" text is a bit abrupt. Usually these messages either tell you what to do instead or tell you what the future behavior will be.

**Comment by csabella** [comment]
@ZackerySpytz, please address @pganssle's comment.  Thanks!

**Comment by csabella** [comment]
@ZackerySpytz, this looks like it was close to being merged.  Please take a look at the last code review.  Thanks!

File: Doc/library/shlex.rst (hunk 1 above)
Comment (ZackerySpytz) [reply Doc/library/shlex.rst]: Okay. Done.

**Review by pganssle — APPROVED** [review]
LGTM. I assume the test failures are unrelated? Happy to merge once we have them green.

**Comment by ZackerySpytz** [comment]
Yes, the CI failures were unrelated (I see errors like `Git checkout failed with exit code: 128`). Everything is green now.

**Comment by pganssle** [comment]
Great, thanks @ZackerySpytz!

**Comment by furkanonder** [comment]
@pganssle The issue seems to be solved. We can close the issue.

**Pull request merged**